### ISSUE TITLE
Wrong result in Buildings.Fluid.HydronicConfigurations.PassiveNetworks.Examples.SingleMixing

### ISSUE THREAD

**Issue by mwetter** [OpenModelica/OpenModelica#10580]
### Description
The model `Buildings.Fluid.HydronicConfigurations.PassiveNetworks.Examples.SingleMixing` (and others in this package that use similar constructs) have wrong results with OpenModelica.

The result below is from Buildings, commit 9b4be2033c1765ad5f518352ee164b8c7d78e775
The error is in the instance `T2SetMod`:
![image](https://user-images.githubusercontent.com/1329080/233166175-dae1bf7b-5b63-4105-bfc5-e7afb4b143b4.png)

The OpenModelica results are as follows:
![image](https://user-images.githubusercontent.com/1329080/233166548-ea222567-43ad-43c7-9507-faf86829dbf6.png)

Note that `index` changes, but the output `y` does not. 
It looks like the line
https://github.com/lbl-srg/modelica-buildings/blob/9b4be2033c1765ad5f518352ee164b8c7d78e775/Buildings/Controls/OBC/CDL/Routing/RealExtractor.mo#L21
is not computed properly.

### Steps to Reproduce

Download Buildings, commit 9b4be2033c1765ad5f518352ee164b8c7d78e775
Simulate `Buildings.Fluid.HydronicConfigurations.PassiveNetworks.Examples.SingleMixing` with dassl, 1E-6.

### Expected Behavior

See above.

### Version and OS

 - OpenModelica Version: OpenModelica 1.22.0~dev-41-g8a5b18f
 - OS: Linux Ubuntu 20.04

**Comment by mwetter** [comment]
It turns out that this work-around gives the right result:
https://github.com/lbl-srg/modelica-buildings/commit/48ad1f65a3d94aab4b32a232b3273304349d875f
![image](https://user-images.githubusercontent.com/1329080/233193900-c2054daf-425b-4493-b3d3-37d73636d5a6.png)

**Comment by casella** [comment]
That's kind of weird, the two formulations should be totally equivalent.

**Comment by rfranke** [comment]
It works with the Cpp runtime after PR #10642. See also #10603 for another example generating different events for C and C++.

**Comment by mahge** [comment]
I am not a 100% sure yet but this seems to be an issue with the `removeSimpleEquations` module. It appears the module constant evaluates the equation when it is in an equation section. It probably should not. 

The library version in question is not available in the OpenModelica package manager (?) so I am working with a MWEs that I think are similar. It would be very helpful if you can provide a MWE for it. Otherwise I will check more and let you know.

**Comment by casella** [comment]
@mahge I tried to come up with an MWE only containing the signal blocks, but if I do so, everything works fine. I attach an MWE package to reproduce the issue (compare SingleMixing with SingleMixingAlgorithm, which uses the tweaked block with the algorithm instead of the equation)
[MWE.zip](https://github.com/OpenModelica/OpenModelica/files/11573326/MWE.zip)

**Comment by mahge** [comment]
> @mahge I tried to come up with an MWE only containing the signal blocks, but if I do so, everything works fine

Yes it was annoying me as well 🙂. One thing I found that made the two (algorithm vs equation) behave different was to have `index` as a parameter with no binding in my MWEs. However, I was not 100% sure if that represented the actual problem here.

I will check your MWE and will let you know.

**Comment by casella** [comment]
> I will check your MWE and will let you know.

It's not such a big deal, it is basically just a copy of the original model, just set up so it depends on the installed version.
